system/jlib/jptree.cpp

```diff
@@ -1469,10 +1469,10 @@ void PTree::appendPropBin(const char *xpath, size32_t size, const void *data)
         appendLocal(size, data, true);
     else if ('[' == *xpath)
     {
-        aindex_t pos = getChildMatchPos(xpath);
-        if ((aindex_t) -1 == pos)
-            throw MakeIPTException(-1, "appendPropBin: qualifier unmatched %s", xpath);
-        appendLocal(size, data, true);
+        IPropertyTree *qualified = queryPropTree(xpath);
+        if (!qualified)
+            throw MakeIPTException(-1, "addProp: qualifier unmatched %s", xpath);
```

Comment:
addPropBin ?

Reply:
Still confusing why adddProp/addPropBin instead of appendProp/appendPropBin

Reply:
will correct.